# Data Wrangling - Capstone Two - King County House Sales

In [1]:
# Import required python libraries
import pandas as pd

In [2]:
# Load 'kc_house_data.csv'
df = pd.read_csv('/Users/mitchmodlich/Desktop/Springboard_Data_Science/Springboard/KingCountyHouseSales/data/raw/kc_house_data.csv')

In [3]:
# View DataFrame head
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [4]:
# Review column names and data types
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

### Column Descriptions:

* id - Unique ID for each sale
* date - Date of sale
* price - Sale price ($)
* bedrooms - Number of bedrooms
* bathrooms -  Number of bathrooms
* sqft_living - Area of interior living space (sq ft)
* sqft_lot - Area of land on lot (sq ft)
* floors - Number of floors
* waterfront - Waterfront location
* view - Rating of property view (0-4)
* condition - Rating of property condition (1-5)
* grade - Rating of house construction and design quality (1-13)
* sqft_above - Area of interior living space above ground level (sq ft)
* sqft_basement - Area of interior living space below ground level (sq ft)
* yr_built - Year house was built
* yr_renovated - Year house was last renovated
* zipcode - Zipcode
* lat - latitude coordinate
* long - longitude coordinate
* sqft_living15 - Average area of interior living space of 15 nearest neighbors
* sqft_lot15 - Average are of lot of 15 nearest neighbors

Some data types should be changed: 

* 'date' to datetime object
* 'yr_built', 'yr_renovated' to string
* 'price' to integer?

In [5]:
# Check for non-integer values in 'price' column
df[df['price'] % 1 == 0]

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

'price' column contains only integer values, so data type should be converted to int64.

In [6]:
# Convert data types

df['date'] = pd.to_datetime(df['date'])

df['yr_built'] = df['yr_built'].astype(str)

df['yr_renovated'] = df['yr_renovated'].astype(str)

df['price'] = df['price'].astype(int)

In [7]:
# Verify data type changes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   id             21613 non-null  int64         
 1   date           21613 non-null  datetime64[ns]
 2   price          21613 non-null  int64         
 3   bedrooms       21613 non-null  int64         
 4   bathrooms      21613 non-null  float64       
 5   sqft_living    21613 non-null  int64         
 6   sqft_lot       21613 non-null  int64         
 7   floors         21613 non-null  float64       
 8   waterfront     21613 non-null  int64         
 9   view           21613 non-null  int64         
 10  condition      21613 non-null  int64         
 11  grade          21613 non-null  int64         
 12  sqft_above     21613 non-null  int64         
 13  sqft_basement  21613 non-null  int64         
 14  yr_built       21613 non-null  object        
 15  yr_renovated   2161

In [8]:
# Check for unique values in 'id' column
df['id'].nunique()

21436

21613 rows, 21436 unique values. Are there duplicate rows?

In [9]:
df[df['id'].duplicated()]

id       date    price  bedrooms  bathrooms  sqft_living  \
94     6021501535 2014-12-23   700000         3       1.50         1580   
314    4139480200 2014-12-09  1400000         4       3.25         4290   
325    7520000520 2015-03-11   240500         2       1.00         1240   
346    3969300030 2014-12-29   239900         4       1.00         1000   
372    2231500030 2015-03-24   530000         4       2.25         2180   
...           ...        ...      ...       ...        ...          ...   
20181  7853400250 2015-02-19   645000         4       3.50         2910   
20613  2724049222 2014-12-01   220000         2       2.50         1000   
20670  8564860270 2015-03-30   502000         4       2.50         2680   
20780  6300000226 2015-05-04   380000         4       1.00         1200   
21581  7853420110 2015-05-04   625000         3       3.00         2780   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
94         5000     1.0           0     0  ...      8        1290   
314       12103     1.0           0     3  ...     11        2690   
325       12092     1.0           0     0  ...      6         960   
346        7134     1.0           0     0  ...      6        1000   
372       10754     1.0           0     0  ...      7        1100   
...         ...     ...         ...   ...  ...    ...         ...   
20181      5260     2.0           0     0  ...      9        2910   
20613      1092     2.0           0     0  ...      7         990   
20670      5539     2.0           0     0  ...      8        2680   
20780      2171     1.5           0     0  ...      7        1200   
21581      6000     2.0           0     0  ...      9        2780   

       sqft_basement  yr_built yr_renovated zipcode      lat     long  \
94               290      1939            0   98117  47.6870 -122.386   
314             1600      1997            0   98006  47.5503 -122.102   
325              280      1922         1984   98146  47.4957 -122.352   
346                0      1943            0   98178  47.4897 -122.240   
372             1080      1954            0   98133  47.7711 -122.341   
...              ...       ...          ...     ...      ...      ...   
20181              0      2012            0   98065  47.5168 -121.883   
20613             10      2004            0   98118  47.5419 -122.271   
20670              0      2013            0   98045  47.4759 -121.734   
20780              0      1933            0   98133  47.7076 -122.342   
21581              0      2013            0   98065  47.5184 -121.886   

       sqft_living15  sqft_lot15  
94              1570        4500  
314             3860       11244  
325             1820        7460  
346             1020        7138  
372             1810        6929  
...              ...         ...  
20181           2910        5260  
20613           1330        1466  
20670           2680        5992  
20780           1130        1598  
21581           2850        6000  

[177 rows x 21 columns]

177 rows have a duplicated 'id' value. What's going on with these rows?

In [10]:
# Set duplicated() method parameter keep=False to retain all duplicates in list for comparison
df[df['id'].duplicated(keep=False)]

id       date    price  bedrooms  bathrooms  sqft_living  \
93     6021501535 2014-07-25   430000         3       1.50         1580   
94     6021501535 2014-12-23   700000         3       1.50         1580   
313    4139480200 2014-06-18  1384000         4       3.25         4290   
314    4139480200 2014-12-09  1400000         4       3.25         4290   
324    7520000520 2014-09-05   232000         2       1.00         1240   
...           ...        ...      ...       ...        ...          ...   
20670  8564860270 2015-03-30   502000         4       2.50         2680   
20779  6300000226 2014-06-26   240000         4       1.00         1200   
20780  6300000226 2015-05-04   380000         4       1.00         1200   
21580  7853420110 2014-10-03   594866         3       3.00         2780   
21581  7853420110 2015-05-04   625000         3       3.00         2780   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
93         5000     1.0           0     0  ...      8        1290   
94         5000     1.0           0     0  ...      8        1290   
313       12103     1.0           0     3  ...     11        2690   
314       12103     1.0           0     3  ...     11        2690   
324       12092     1.0           0     0  ...      6         960   
...         ...     ...         ...   ...  ...    ...         ...   
20670      5539     2.0           0     0  ...      8        2680   
20779      2171     1.5           0     0  ...      7        1200   
20780      2171     1.5           0     0  ...      7        1200   
21580      6000     2.0           0     0  ...      9        2780   
21581      6000     2.0           0     0  ...      9        2780   

       sqft_basement  yr_built yr_renovated zipcode      lat     long  \
93               290      1939            0   98117  47.6870 -122.386   
94               290      1939            0   98117  47.6870 -122.386   
313             1600      1997            0   98006  47.5503 -122.102   
314             1600      1997            0   98006  47.5503 -122.102   
324              280      1922         1984   98146  47.4957 -122.352   
...              ...       ...          ...     ...      ...      ...   
20670              0      2013            0   98045  47.4759 -121.734   
20779              0      1933            0   98133  47.7076 -122.342   
20780              0      1933            0   98133  47.7076 -122.342   
21580              0      2013            0   98065  47.5184 -121.886   
21581              0      2013            0   98065  47.5184 -121.886   

       sqft_living15  sqft_lot15  
93              1570        4500  
94              1570        4500  
313             3860       11244  
314             3860       11244  
324             1820        7460  
...              ...         ...  
20670           2680        5992  
20779           1130        1598  
20780           1130        1598  
21580           2850        6000  
21581           2850        6000  

[353 rows x 21 columns]

It appears that the 'id' column contains a unique value per lot/house. This value does not change if the house is sold again, which seems to have occurred for 177 unique houses over the time period from May 2014 through May 2015.

The index assigned by pandas during the loading process will be retained to have a unique identifier column for each row. None of the duplicated rows will be dropped.

In [11]:
# Find number of unique date values
df['date'].nunique()

372

The timespan covered is more than one calendar year, and houses were sold on 372 unique dates over this period.

In [12]:
# Check simple summary statistics
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement       zipcode           lat  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045  98077.939805     47.560053   
std        1.175459    828.090978     442.575043     53.505026      0.138564   
min        1.000000    290.000000       0.000000  98001.000000     47.155900   
25%        7.000000   1190.000000       0.000000  98033.000000     47.471000   
50%        7.000000   1560.000000       0.000000  98065.000000     47.571800   
75%        8.000000   2210.000000     560.000000  98118.000000     47.678000   
max       13.000000   9410.000000    4820.000000  98199.000000     47.777600   

               long  sqft_living15     sqft_lot15  
count  21613.000000   21613.000000   21613.000000  
mean    -122.213896    1986.552492   12768.455652  
std        0.140828     685.391304   27304.179631  
min     -122.519000     399.000000     651.000000  
25%     -122.328000    1490.000000    5100.000000  
50%     -122.230000    1840.000000    7620.000000  
75%     -122.125000    2360.000000   10083.000000  
max     -121.315000    6210.000000  871200.000000

An inspection of these summary statistics shows that some rows have zeroes for bedrooms and bathrooms values.

In [13]:
df[df['bedrooms'] == 0]

id       date    price  bedrooms  bathrooms  sqft_living  \
875    6306400140 2014-06-12  1095000         0       0.00         3064   
3119   3918400017 2015-02-05   380000         0       0.00         1470   
3467   1453602309 2014-08-05   288000         0       1.50         1430   
4868   6896300380 2014-10-02   228000         0       1.00          390   
6994   2954400190 2014-06-24  1295650         0       0.00         4810   
8477   2569500210 2014-11-17   339950         0       2.50         2290   
8484   2310060040 2014-09-25   240000         0       2.50         1810   
9773   3374500520 2015-04-29   355000         0       0.00         2460   
9854   7849202190 2014-12-23   235000         0       0.00         1470   
12653  7849202299 2015-02-18   320000         0       2.50         1490   
14423  9543000205 2015-04-13   139950         0       0.00          844   
18379  1222029077 2014-10-29   265000         0       0.75          384   
19452  3980300371 2014-09-26   142000         0       0.00          290   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
875        4764     3.5           0     2  ...      7        3064   
3119        979     3.0           0     2  ...      8        1470   
3467       1650     3.0           0     0  ...      7        1430   
4868       5900     1.0           0     0  ...      4         390   
6994      28008     2.0           0     0  ...     12        4810   
8477       8319     2.0           0     0  ...      8        2290   
8484       5669     2.0           0     0  ...      7        1810   
9773       8049     2.0           0     0  ...      8        2460   
9854       4800     2.0           0     0  ...      7        1470   
12653      7111     2.0           0     0  ...      7        1490   
14423      4269     1.0           0     0  ...      7         844   
18379    213444     1.0           0     0  ...      4         384   
19452     20875     1.0           0     0  ...      1         290   

       sqft_basement  yr_built yr_renovated zipcode      lat     long  \
875                0      1990            0   98102  47.6362 -122.322   
3119               0      2006            0   98133  47.7145 -122.356   
3467               0      1999            0   98125  47.7222 -122.290   
4868               0      1953            0   98118  47.5260 -122.261   
6994               0      1990            0   98053  47.6642 -122.069   
8477               0      1985            0   98042  47.3473 -122.151   
8484               0      2003            0   98038  47.3493 -122.053   
9773               0      1990            0   98031  47.4095 -122.168   
9854               0      1996            0   98065  47.5265 -121.828   
12653              0      1999            0   98065  47.5261 -121.826   
14423              0      1913            0   98001  47.2781 -122.250   
18379              0      2003            0   98070  47.4177 -122.491   
19452              0      1963            0   98024  47.5308 -121.888   

       sqft_living15  sqft_lot15  
875             2360        4000  
3119            1470        1399  
3467            1430        1650  
4868            2170        6000  
6994            4740       35061  
8477            2500        8751  
8484            1810        5685  
9773            2520        8050  
9854            1060        7200  
12653           1500        4675  
14423           1380        9600  
18379           1920      224341  
19452           1620       22850  

[13 rows x 21 columns]

In [14]:
df[df['bathrooms'] == 0]

id       date    price  bedrooms  bathrooms  sqft_living  \
875    6306400140 2014-06-12  1095000         0        0.0         3064   
1149   3421079032 2015-02-17    75000         1        0.0          670   
3119   3918400017 2015-02-05   380000         0        0.0         1470   
5832   5702500050 2014-11-04   280000         1        0.0          600   
6994   2954400190 2014-06-24  1295650         0        0.0         4810   
9773   3374500520 2015-04-29   355000         0        0.0         2460   
9854   7849202190 2014-12-23   235000         0        0.0         1470   
10481   203100435 2014-09-18   484000         1        0.0          690   
14423  9543000205 2015-04-13   139950         0        0.0          844   
19452  3980300371 2014-09-26   142000         0        0.0          290   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
875        4764     3.5           0     2  ...      7        3064   
1149      43377     1.0           0     0  ...      3         670   
3119        979     3.0           0     2  ...      8        1470   
5832      24501     1.0           0     0  ...      3         600   
6994      28008     2.0           0     0  ...     12        4810   
9773       8049     2.0           0     0  ...      8        2460   
9854       4800     2.0           0     0  ...      7        1470   
10481     23244     1.0           0     0  ...      7         690   
14423      4269     1.0           0     0  ...      7         844   
19452     20875     1.0           0     0  ...      1         290   

       sqft_basement  yr_built yr_renovated zipcode      lat     long  \
875                0      1990            0   98102  47.6362 -122.322   
1149               0      1966            0   98022  47.2638 -121.906   
3119               0      2006            0   98133  47.7145 -122.356   
5832               0      1950            0   98045  47.5316 -121.749   
6994               0      1990            0   98053  47.6642 -122.069   
9773               0      1990            0   98031  47.4095 -122.168   
9854               0      1996            0   98065  47.5265 -121.828   
10481              0      1948            0   98053  47.6429 -121.955   
14423              0      1913            0   98001  47.2781 -122.250   
19452              0      1963            0   98024  47.5308 -121.888   

       sqft_living15  sqft_lot15  
875             2360        4000  
1149            1160       42882  
3119            1470        1399  
5832             990       22549  
6994            4740       35061  
9773            2520        8050  
9854            1060        7200  
10481           1690       19290  
14423           1380        9600  
19452           1620       22850  

[10 rows x 21 columns]

Why do some houses have 0 bedrooms and 0 bathrooms? The other columns are populated with data for these rows. These rows may need to be dropped.

In [15]:
# Check sample row with no bedrooms
df.iloc[875]

id                        6306400140
date             2014-06-12 00:00:00
price                        1095000
bedrooms                           0
bathrooms                        0.0
sqft_living                     3064
sqft_lot                        4764
floors                           3.5
waterfront                         0
view                               2
condition                          3
grade                              7
sqft_above                      3064
sqft_basement                      0
yr_built                        1990
yr_renovated                       0
zipcode                        98102
lat                          47.6362
long                        -122.322
sqft_living15                   2360
sqft_lot15                      4000
Name: 875, dtype: object

In [16]:
# Check for null values
df.isna().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

In [17]:
# Create list of index values where 'bathroom' or 'bedroom' value = 0
df_ind = df[(df['bathrooms'] == 0) | (df['bedrooms'] == 0)].index

In [18]:
df_ind

Int64Index([  875,  1149,  3119,  3467,  4868,  5832,  6994,  8477,  8484,
             9773,  9854, 10481, 12653, 14423, 18379, 19452],
           dtype='int64')

In [19]:
# Drop df_ind rows from DataFrame
df_clean = df.drop(df_ind)

In [20]:
# Confirm rows dropped
df_clean

id       date   price  bedrooms  bathrooms  sqft_living  \
0      7129300520 2014-10-13  221900         3       1.00         1180   
1      6414100192 2014-12-09  538000         3       2.25         2570   
2      5631500400 2015-02-25  180000         2       1.00          770   
3      2487200875 2014-12-09  604000         4       3.00         1960   
4      1954400510 2015-02-18  510000         3       2.00         1680   
...           ...        ...     ...       ...        ...          ...   
21608   263000018 2014-05-21  360000         3       2.50         1530   
21609  6600060120 2015-02-23  400000         4       2.50         2310   
21610  1523300141 2014-06-23  402101         2       0.75         1020   
21611   291310100 2015-01-16  400000         3       2.50         1600   
21612  1523300157 2014-10-15  325000         2       0.75         1020   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
0          5650     1.0           0     0  ...      7        1180   
1          7242     2.0           0     0  ...      7        2170   
2         10000     1.0           0     0  ...      6         770   
3          5000     1.0           0     0  ...      7        1050   
4          8080     1.0           0     0  ...      8        1680   
...         ...     ...         ...   ...  ...    ...         ...   
21608      1131     3.0           0     0  ...      8        1530   
21609      5813     2.0           0     0  ...      8        2310   
21610      1350     2.0           0     0  ...      7        1020   
21611      2388     2.0           0     0  ...      8        1600   
21612      1076     2.0           0     0  ...      7        1020   

       sqft_basement  yr_built yr_renovated zipcode      lat     long  \
0                  0      1955            0   98178  47.5112 -122.257   
1                400      1951         1991   98125  47.7210 -122.319   
2                  0      1933            0   98028  47.7379 -122.233   
3                910      1965            0   98136  47.5208 -122.393   
4                  0      1987            0   98074  47.6168 -122.045   
...              ...       ...          ...     ...      ...      ...   
21608              0      2009            0   98103  47.6993 -122.346   
21609              0      2014            0   98146  47.5107 -122.362   
21610              0      2009            0   98144  47.5944 -122.299   
21611              0      2004            0   98027  47.5345 -122.069   
21612              0      2008            0   98144  47.5941 -122.299   

       sqft_living15  sqft_lot15  
0               1340        5650  
1               1690        7639  
2               2720        8062  
3               1360        5000  
4               1800        7503  
...              ...         ...  
21608           1530        1509  
21609           1830        7200  
21610           1020        2007  
21611           1410        1287  
21612           1020        1357  

[21597 rows x 21 columns]

In [21]:
# Write df_clean to csv file
with open('/Users/mitchmodlich/Desktop/Springboard_Data_Science/Springboard/KingCountyHouseSales/data/interim/kc_house_data_clean.csv', 'w') as file:
    df_clean.to_csv(file)